rfc-175-frontend-fewer-apps.md

```diff
@@ -0,0 +1,254 @@
+
+### Steps as part of a merge
+We anticipate that any merge will likely involve some or all of the following steps.
+
+* Identify any issues with existing code
+    * size these issues
+    * either fix issues or record them as tech debt - we need to avoid trying to fix everything and concentrate on progressing consolidation
+* Replicate the rendering from the candidate to the host
+    * preserve the git history of the candidate app - add a permalink to the files being consolidated in the commit in the host app.
+* Change content api
+    * change the rendering app for the document type in the publishing app.
+    * republish content
+    * republish any special routes (things that don’t have a publishing app, like the homepage) - any special routes should be moved into special route publisher (we have an outstanding tech debt card relating to this)
+* Optimise the frontend code
+    * remove duplication
+* Remove the routes from the candidate
+* [Retire the candidate] once all routes have been removed
+
+### Expected issues
+Merging these four apps is not without difficulties and we acknowledge that a single app will present problems that our multiple apps do not.
+
+* A Rails update for one larger app might be more complex than spread across individual apps.
+* It won’t be possible to scale targeted parts of the stack (i.e. one of the existing apps) for expected increases in traffic.
+* It’s only possible to test one branch of code in a test environment at once. If there’s only one repo for these four apps, only one change can be tested at a time.
+* It will be harder to find the pull request history for the retired repositories.
+
+In addition, we will need to address the following issues:
+
+* The test suite will take longer due to more tests.
+    * could potentially parallelise in CI
+    * will need to consider approach for developer machines
+* Need to reconfigure Kubernetes deployment. We should be able to drop the total number of pods.
+    * optimise the number of pods needed in draft and production
+    * currently in production there are
+        * 4 pods for `frontend`
+        * 6 for `government-frontend`
+        * 4 for `collections`
+        * 9 for `finder-frontend`
+* Resolve the question around environment testing
+    * rather than testing on integration, we should still be able to test multiple branches using preview apps
+    * would need to get new search working in the preview apps
+
+### After merging
+Once a candidate has been merged, the following steps would be needed.
+
+* Recalibrate Kubernetes deployment
+* Move components from govuk_publishing_components into the app if only being used in the frontend
+    * this information can already be found in our component auditing tools
+    * need to consider which ones are used by publishing applications
+* Update component auditing
+* Decide how to handle app ownership and maintenance and what that means now we have fewer apps.
+* Consider renaming `frontend`
```

Comment:
Time to reclaim govuk-frontend 💪

Reply:
Or could fall back to the old suggestion from https://github.com/alphagov/govuk-rfcs/blob/main/rfc-095-long-term-future-apps.md of `content-frontend`

Reply:
The hardest problem in computer science, naming things!

Reply:
This is the kind of thing that should probably be put to a vote. I'll have to veto "frontendy-mcfrontend-face" though 😉 .